src/gramps.js

```diff
@@ -165,9 +165,5 @@ export function prepare({
 }
 
 export default function gramps(...args) {
-  const options = prepare(...args);
-  return req => ({
```

Comment:
I'm pretty sure this callback is required for e.g. adding data from the request into the context. Does Apollo Server 2.0 remove support for this?

Removing it would be a breaking change (which would be the case anyways, given that this is a major dependency bump, I guess.).

@ryanomackey will be the final call on whether or not this merges. Just wanted to flag this.

Reply:
@jlengstorf In the new version the context takes a callback with `req` for its value.
```
new ApolloServer({
	typeDefs,
	resolvers,
	context: ({ req }) => ({
		authScope: getScope(req.headers.authorization)
	}),
});
```
No need to wrap the whole graphql-server options in a callback.

Reply:
Hey, thanks @shalkam. Definitely a breaking change, so we'll need to include that in the commit message. See [the semantic release docs](https://github.com/semantic-release/semantic-release#how-does-it-work) for details.

And we do still need some implementation of that callback. I took a look, and something like this would work nicely:

```
export default function gramps(...args) {
  const options = prepare(...args);

  return {
    ...options,
    context: ({ req }) => options.context(req),
  }
}
```

Let me know what y'all think.